### PR TITLE
Have the bot label issues by type

### PR DESCRIPTION
**Pull request by strugee**
What the title says. Open questions:

* `ruleset enhancement` as an issue type seems like a bit of a mouthful - maybe it should be "ruleset feature" or something instead, I dunno
* There's a `ruleset` label but it doesn't seem to be used much. Should the bot always label ruleset issues with this label?
* There's no general `code` label so I used `bug` and `enhancement` which, conveniently, there are already types for (`code issue` and `feature request`, respectively). Does that seem right?

All tests passing locally.

As usual this contains an unrelated commit (removing an outdated TODO) that didn't seem worth bothering with a separate PR for.

Ref #14616

/cc @Hainish

### CODE REVIEW

**Comment by strugee** [comment]
@Hainish ping, when you get a chance

**Comment by strugee** [comment]
@Hainish ping :-)

**Comment by strugee** [comment]
@Hainish ping

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 [//]: # (Leave in the line that best describes the issue you are reporting.)
 
 Type: ruleset issue
+Type: ruleset enhancement
```
Comment (Hainish) [inline .github/ISSUE_TEMPLATE.md]: I think @jeremyn's comment in https://github.com/EFForg/https-everywhere/issues/13890#issuecomment-351230054 is fair: we should probably reduce the overhead and only have a `ruleset issue` and `new ruleset` sub-tier label, both of which would get the `ruleset` label.

Comment (strugee) [reply .github/ISSUE_TEMPLATE.md]: @Hainish whoops, this fell off my radar somehow. Fixed.

**Comment by Bisaloo** [comment]
@strugee, [this recent email from the dev mailing list](https://lists.eff.org/pipermail/https-everywhere/2018-March/002772.html) is relevant here: the `ruleset` label has been deleted.

The `ruleset issue` issues should probably now use `ruleset-bug`.

**Comment by strugee** [comment]
I emailed the list to solicit feedback about this: https://lists.eff.org/pipermail/https-everywhere/2018-April/002777.html

**Comment by strugee** [comment]
I emailed @Hainish this evening and realized that once again I'd forgotten about this PR! Since I never got any explicit feedback on the mailing list I just went with what was sensible while not adding any more issue types. See the commit message in 5a9dce2839 for details.

This is ready for review again.

**Comment by strugee** [comment]
I just rebased this on top of `master` too.

**Comment by Bisaloo** [comment]
The labels look fine to me. Thank you!

**Comment by strugee** [comment]
@Bisaloo great! What do you think of the note to ping me? Too noisy?

One thing that wouldn't be too hard to do is make it so that the bot e.g. only commented this once in a given thread. It would be more difficult but we should also be able to do behaviors like "every other comment in a thread". I've included information so that we can do that now or later if desired.

**Comment by Bisaloo** [comment]
I'm sorry, I'm not sure what you're referring to.

**Comment by strugee** [comment]
@Bisaloo ahaha, so sorry! I mixed up the PR where I made that change. I thought it was in this one, but it was actually in #15439 instead. My bad :D

**Comment by Hainish** [comment]
Thank you!

**Comment by strugee** [comment]
@Hainish my pleasure!

**Comment by strugee** [comment]
And do let me know if it's helpful or if you run into problems :)

**Pull request merged**